### PR TITLE
Fix links and add test to glew

### PR DESCRIPTION
**Pull request by PucklaMotzer09**
This PR fixes the following
+ Adds test
+ Correct links when static linking and dynamic linking
  + glew32s and glew32
+ Correct Preprocessor Macros when static linking and dynamic linking
+ Correct syslinks

### CODE REVIEW

**Comment by PucklaMotzer09** [comment]
The linux build is failing because ```libGL``` cannot be found. This can be fixed by installing a OpenGL library:
```
sudo apt-get install libgl1-mesa-dev
```

**Comment by waruqi** [comment]
> The linux build is failing because `libGL` cannot be found. This can be fixed by installing a OpenGL library:
> 
> ```
> sudo apt-get install libgl1-mesa-dev
> ```

In order to pass the test, you can temporarily add it in linux.yml

**Comment by PucklaMotzer09** [comment]
It seems that the MinGW build is building for MacOSX. The arguments are the exact same. There even is ```-framework OpenGL``` which only gets added when the platform is ```macosx```

**Comment by waruqi** [comment]
> It seems that the MinGW build is building for MacOSX. The arguments are the exact same. There even is `-framework OpenGL` which only gets added when the platform is `macosx`

? Yes, what's the problem now?

**Comment by PucklaMotzer09** [comment]
So this is expected behaviour? I thought that it would compile more like the windows version like it used to in all the other libraries.

**Comment by waruqi** [comment]
> So this is expected behaviour? I thought that it would compile more like the windows version like it used to in all the other libraries.

On macos, we can also use
mingw sdk to compile windows libraries.

**Comment by PucklaMotzer09** [comment]
Should the Installation step of linux stay or what are we going to do about it?

**Comment by waruqi** [comment]
And we can alse add msys2/mingw on windows for ci.

like https://github.com/xmake-io/xmake/blob/master/.github/workflows/msys2_mingw.yml

**Comment by waruqi** [comment]
> Should the Installation step of linux stay or what are we going to do about it?

If the tests have all passed, you can revert linux.yml and I will merge it

**Comment by PucklaMotzer09** [comment]
I removed the Installation step. You would have to add the following command
```
sudo apt-get install -y libgl1-mesa-dev libglu1-mesa-dev
```

File: packages/g/glew/xmake.lua [hunk 1]
```diff
@@ -11,8 +11,29 @@ package("glew")
...
+    on_load(function (package)
+        if package:config("shared") then
+            package:add("defines", "GLEW_BUILD")
+            if package:is_plat("windows", "mingw") then
+                package:add("links", "glew32")
+            end
+        else
+            package:add("defines", "GLEW_STATIC")
+            if package:is_plat("windows", "mingw") then
+                package:add("links", "glew32s")
```
Comment (waruqi) [inline packages/g/glew/xmake.lua:34]: and here. syslinks

Comment (PucklaMotzer09) [reply packages/g/glew/xmake.lua:34]: The syslinks are always the same. So I added them above using ```add_syslinks```

Comment (waruqi) [reply packages/g/glew/xmake.lua:34]: You can use add_syslinks to instead of `package:add("links")` or use `package:add("syslinks")`

Comment (PucklaMotzer09) [reply packages/g/glew/xmake.lua:34]: But glew32 is not a syslink.

Comment (waruqi) [reply packages/g/glew/xmake.lua:34]: Oh, glew32 is not. my mistake.

**Comment by waruqi** [comment]
I have merged it.

**Comment by waruqi** [comment]
> It seems that the MinGW build is building for MacOSX. The arguments are the exact same. There even is `-framework OpenGL` which only gets added when the platform is `macosx`

I have added msys/mingw ci and updated xmake/dev to support for using cmake/autotools on msys/mingw.

**Pull request merged**